File: Element with left side smaller and right side greater - GFG/element-with-left-side-smaller-and-right-side-greater.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

int findElement(int arr[], int n);

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, i;
        cin >> n;
        int a[n];
        for (i = 0; i < n; i++) cin >> a[i];
        cout << findElement(a, n) << endl;
    }
    return 0;
}
// } Driver Code Ends


int findElement(int arr[], int n) {
    int i,v,k;
    int x=0;
    v=arr[0];
    k=0;
    for(i=1;i<n;i++)
    {
        if(arr[i]<v)
        {
            x=0;
            if(i==n-1)
            {
                return(-1);
            }
        }
        else if(arr[i]>=v && x==0)
        {
            x=1;
            v=arr[i];
            k=i;
        }
    }
    if(k==n-1)
    {
        return(-1);
    }
    return(v);
}

File: Element with left side smaller and right side greater - GFG/element-with-left-side-smaller-and-right-side-greater.py
#User function Template for python3

def findElement( arr, n):
    l=[]
    l.append(arr[0])
    a=0
    b=0
    c=-1
    for i in range(1,len(arr)):
        if(i!=len(arr)-1 and arr[i]>=l[a]):
            while(len(l)>0 and arr[i]>=l[a]):
                a-=1
                l.pop()
            if(len(l)==0 and b==0):
                c=arr[i]
                b=1
            l.append(arr[i])
            a+=1
        elif(arr[i]<c):
            b=0
            c=-1
            l.append(arr[i])
            a+=1
    return(c)
            
        
    
    
    


#{ 
#  Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        print(findElement(a, n))

        T -= 1


if __name__ == "__main__":
    main()





    
# } Driver Code Ends